enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -1077,10 +1071,12 @@ ReferenceGrant was introduced as a safeguard against these types of attacks by r
 explicit permission from the target object's owner. The ability to do cross-namespace
 references in OSSM 2.3 without any safeguards is a risk; however, OSSM 2.4's requirement
 for ReferenceGrants mitigates this risk.
+This is why it is important to fulfill the goal that ReferenceGrants may be configured by application developers.  Cluster admins or admins will likely have to add new listeners that listen to all namespaces by default.
```

Comment:
Sorry, this can be removed.  I misunderstood ReferenceGrants, conflating them with allowedNamespaces on listeners.  OpenShift developers won't be expecting ReferenceGrants, but it could be a desirable feature for selling Gateway API.

Reply:
And I'm conflating allowedNamespace with allowedRoutes.namespaces.

Reply:
It now occurs to me that we'll have to ensure allowedRoutes.namespaces needs to be set to All.  The default is Same.  This is the important note for shared vs dedicated Gateways.

Reply:
> Sorry, this can be removed. I misunderstood ReferenceGrants, conflating them with allowedNamespaces on listeners. OpenShift developers won't be expecting ReferenceGrants, but it could be a desirable feature for selling Gateway API.

SGTM

```suggestion
```

> It now occurs to me that we'll have to ensure allowedRoutes.namespaces needs to be set to All for shared Gateways and Same for dedicated Gateways. The default is Same. This is the important note for shared vs dedicated Gateways.

Ok yes yes. This I think clears up some confusion I was having about the notion of "allowing all namespaces by default" we were talking about earlier, just a misunderstanding.

I'm adding some additional notes to the shared gateways about this to clarify :+1: